frontend/components/buttons/RevealButton/RevealButton.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import Button from "components/buttons/Button";
+import TooltipWrapper from "components/TooltipWrapper";
+
+export interface IRevealButtonProps {
+  showBoolean: boolean;
+  baseClass: string;
+  hideString: string;
+  showString: string;
+  caratBefore?: boolean;
+  caratAfter?: boolean;
```

Comment:
If we only have 2 choices, let's combine it as one. So by default we can make the caret after the text, then use `caretBefore` as the boolean to change the position.

Reply:
The observer view show/hide SQL currently uses neither, which I think is stylistically unsound so I added a carat before.

@mike-j-thomas let us know if there should always be carats. If so, we can combine these keys and fix Figma for observer view show/hide SQL.

Reply:
If we need 3 options, then I think one prop as a string would work best so it's one less prop devs have to remember. We could do `caretPosition?: "none" | "before" | "after"`. Then, in the component's props destructure object, we can default `caretPosition = "none"`.